tests/components/numato/test_numato.py

```diff
@@ -0,0 +1,325 @@
+    assert numato_fixture.devices[0].values[5] == 1
+    await async_turn_on(hass, "switch.numato_switch_mock_port6")
+    assert numato_fixture.devices[0].values[6] == 1
+    await async_turn_off(hass, "switch.numato_switch_mock_port5")
+    assert numato_fixture.devices[0].values[5] == 0
+    await async_turn_off(hass, "switch.numato_switch_mock_port6")
+    assert numato_fixture.devices[0].values[6] == 0
+
+    numato_fixture.devices[0].mockup_inject_notification(2, 1)
+    await hass.async_block_till_done()
+    state = hass.states.get("binary_sensor.numato_binary_sensor_mock_port2")
+    assert state.state == "on"
+
+
+@pytest.mark.asyncio
+async def test_failing_hass_operations(hass, numato_fixture, monkeypatch):
+    """Test regular operations from within Home Assistant."""
+    assert await async_setup_component(hass, "numato", NUMATO_CFG)
+
+    await hass.async_block_till_done()  # wait until services are registered
+    monkeypatch.setattr(numato_fixture.devices[0], "write", mockup_raise)
+    await async_turn_on(hass, "switch.numato_switch_mock_port5")
+    assert not numato_fixture.devices[0].values[5]
+    await async_turn_on(hass, "switch.numato_switch_mock_port6")
+    assert not numato_fixture.devices[0].values[6]
+    await async_turn_off(hass, "switch.numato_switch_mock_port5")
+    assert not numato_fixture.devices[0].values[5]
+    await async_turn_off(hass, "switch.numato_switch_mock_port6")
+    assert not numato_fixture.devices[0].values[6]
+
+
+@pytest.mark.asyncio
+async def test_failing_async_added_to_hass(hass, numato_fixture, monkeypatch):
+    """Test condition when a sensor update fails."""
+    monkeypatch.setattr(numato_fixture.NumatoDeviceMock, "setup", mockup_raise)
+    assert await async_setup_component(hass, "numato", NUMATO_CFG)
+    await hass.async_block_till_done()  # wait until services are registered
+
+
+@pytest.mark.asyncio
+async def test_failing_sensor_update(hass, numato_fixture, monkeypatch):
+    """Test condition when a sensor update fails."""
+    monkeypatch.setattr(numato_fixture.NumatoDeviceMock, "adc_read", mockup_raise)
+    assert await async_setup_component(hass, "numato", NUMATO_CFG)
+    await hass.async_block_till_done()  # wait until services are registered
+
+
+@pytest.mark.asyncio
+async def test_hass_numato_api(hass, numato_fixture):
+    """Test regular device access."""
+    assert await async_setup_component(hass, "numato", NUMATO_CFG)
+    api = hass.data["numato"]["api"]
+    # regular operations
+    api.read_adc_input(0, 1)
+    api.read_input(0, 2)
+    api.write_output(0, 2, 1)
+
+    def gen_callback(expected_port, expected_value):
+        def cb_mockup(port, value):
+            assert port == expected_port
+            assert value == expected_value
+
+        return cb_mockup
+
+    api.edge_detect(0, 2, gen_callback(2, 1))
+    numato_fixture.devices[0].mockup_inject_notification(2, 1)
+    api.edge_detect(0, 2, gen_callback(2, 0))
```

Comment:
After calling the callback we want to check the state of the binary sensor. That's what this test should do.

Reply:
This is already done in `test_regular_hass_operations`.

Reply:
Ok, then we don't need this test.